config/tectonic-node-controller/config.go

```diff
@@ -0,0 +1,40 @@
+package tnc
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// Kind is the TypeMeta.Kind for the TNC config.
+	Kind = "TectonicNodeControllerConfig"
+	// APIVersion is the TypeMeta.APIVersion for the TNC config.
+	APIVersion = "v1"
+)
+
+// TectonicNodeControllerConfig is the config structure TNC to generate NodeConfigs
+type TectonicNodeControllerConfig struct {
+	metav1.TypeMeta       `json:",inline"`
+	HTTPProxy             string `json:"HTTPProxy"`
+	HTTPSProxy            string `json:"HTTPSProxy"`
+	NoProxy               string `json:"NoProxy"`
+	KubeletImageURL       string `json:"KubeletImageURL"`
+	KubeletImageTag       string `json:"KubeletImageTag"`
+	IscsiEnabled          string `json:"IscsiEnabled"`
+	KubeconfigFetchCmd    string `json:"KubeconfigFetchCmd"`
```

Comment:
What is this for? This seems like a very odd / out of place parameter. I'm really hoping we can remove this (but it might be out of our control?

@yifan-gu if this one is harder to remove, please just open a follow-up issue to track getting this removed